src/main/java/org/wso2/carbon/module/swiftiso20022/utils/JsonToMt940Utils.java

```diff
@@ -0,0 +1,326 @@
+     *
+     * @param balanceDetails   Balance Details
+     * @param fieldName        Balance Field Name
+     * @return  Return ErrorModel if the balance is invalid.
+     */
+    public static ErrorModel validateBalanceDetails(BalanceModel balanceDetails, String fieldName) {
+
+        if (!isValidDateFormat(balanceDetails.getDate())) {
+            return new ErrorModel(ConnectorConstants.ERROR_T50,
+                    String.format(ConnectorConstants.ERROR_BAL_DATE_INVALID, fieldName));
+        }
+
+        if (!ValidatorUtils.isValidCurrency(balanceDetails.getCurrency())) {
+            return new ErrorModel(ConnectorConstants.ERROR_T52,
+                    String.format(ConnectorConstants.ERROR_CURRENCY_CODE_INVALID,
+                            ConnectorConstants.OPENING_BALANCE));
+        }
+
+        ErrorModel error = ValidatorUtils.validateAmountLength(balanceDetails.getBalanceAmount(),
+                fieldName);
+        if (error.isError()) {
+            return error;
+        }
+
+        if (balanceDetails.getIndicator() == null || balanceDetails.getIndicator().isBlank() ||
+                !(ConnectorConstants.DEBIT.equals(balanceDetails.getIndicator()) ||
+                ConnectorConstants.CREDIT.equals(balanceDetails.getIndicator()))) {
+            return new ErrorModel(ConnectorConstants.ERROR_T51, ConnectorConstants.ERROR_BAL_IND_INVALID);
+        }
+
+        if (balanceDetails.getStatementType() != null && !balanceDetails.getStatementType().isBlank()) {
+            if (!(ConnectorConstants.CURRENT_STATEMENT_TYPE.equals(balanceDetails.getStatementType()) ||
+                    ConnectorConstants.LAST_STATEMENT_TYPE.equals(balanceDetails.getStatementType()))) {
+                return new ErrorModel(ConnectorConstants.ERROR_T51, ConnectorConstants.ERROR_INVALID_STATEMENT_TYPE);
+            }
+        }
+
+        return new ErrorModel();
+    }
+
+    /**
+     * Method to validate the date format.
+     *
+     * @param dateTime  Date time to validate
+     * @return          True if the date format is valid, else false
+     */
+    public static boolean isValidDateFormat(String dateTime) {
+
+        if (dateTime == null || dateTime.isBlank()) {
+            return false;
+        }
+        DateFormat formatter = new SimpleDateFormat(ConnectorConstants.DATE_TIME_FORMAT);
+
+        try {
+            formatter.parse(dateTime);
+            return true;
+        } catch (ParseException e) {
+            return false;
+        }
+    }
+
+    /**
+     * Method to validate the transaction type.
+     *
+     * @param transactionType  Transaction type to validate
+     * @return                 True if the transaction type is valid, else false
+     */
+    public static ErrorModel isValidTransactionType(String transactionType) {
+        ErrorModel errorModel = new ErrorModel();
+        if (!transactionType.startsWith("S") && !transactionType.startsWith("N") &&
```

Comment:
We can move these to constants since used in multiple places.

Reply:
Fixed in https://github.com/wso2-extensions/mediation-swift-iso20022-module/pull/2/commits/18ec29ed3926964e6b999c6fd5d385b24c1f8fa5